docs/providers/api-version.tsx

```diff
@@ -0,0 +1,69 @@
+import { useRouter } from 'next/router';
+import React, {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+import { isVersionedUrl, getVersionFromUrl, replaceVersionInUrl } from '~/common/utilities';
+
+type Props = PropsWithChildren<object>;
+
+const context = createContext({
+  version: 'latest',
+  setVersion: (newVersion: string) => console.error('ApiVersion context provider not found'),
+});
+
+export function ApiVersionProvider(props: Props) {
+  const router = useRouter();
+  const [version, setVersion] = useState('latest');
+
+  const onVersionChange = useCallback((newVersion: string) => {
+    if (isVersionedUrl(router.pathname)) {
```

Comment:
We could also check if the version is valid, but I like to think our code only uses valid versions